### ISSUE TITLE
FormatException must store the offending value

### ISSUE THREAD

**Issue by yecril71pl** [dotnet/runtime#39270]
### Description

```cs
string offendingValue = "?";
try { 
int.Parse (offendingValue); }
 catch (System.FormatException x) {
 Assert.AreEqual (offendingValue, x.Data["Value"], "FormatException must store the offending value."); }
```
### Configuration

.NETCoreApp,Version: v2.1
OS:                             Microsoft Windows 10.0.18363
architecture: x64
### Other information

Error Message:
 Assert.AreEqual failed. Expected:<?>. Actual:<(null)>. FormatException must store the offending value.

**Comment by Gnbrkm41** [comment]
This is a feature request, correct? I don't remember the exceptions actually containing the invalid value.

**Comment by yecril71pl** [comment]
I tried to report a feature request but there is no such option in the project.

**Comment by Gnbrkm41** [comment]
API proposal might work, but I don't it really matters >_>.

**Comment by yecril71pl** [comment]
This request does not introduce any change into the API.

**Comment by Gnbrkm41** [comment]
I know, the template doesn't really matter (in fact you can start with an empty one if you wish). I just wanted to confirm that this isn't a bug report of some sort, the example seemed like one for me personally 🙂.

**Comment by yecril71pl** [comment]
The example was a shot at TDD.

**Comment by joperezr** [comment]
@yecril71pl is correct that this doesn't represent an API proposal as it would just be a behavior change so I marked it as enhancement proposal.

@stephentoub do we have the precedent of adding things into the exception's property bag when it makes sense to do so?

@yecril71pl can you elaborate a bit more on which type of scenario you were thinking off here? I ask because most of the time when developers don't control the value getting passed in to Parse they would call TryParse instead such as:

```c#
public void ParseSomeNumber(string value)
{
  if (int.TryParse(value, out int theNumber))
  {
    //...
  }
  else
  {
    //.. log an error here or throw your own exception with value, which you do have here in scope
  }
}
```

so I honestly don't see many people relying on this and start calling Parse instead of TryParse in most cases.

**Comment by yecril71pl** [comment]
It is for PowerShell mainly, scripters are unlikely to call TryParse.

**Comment by stephentoub** [comment]
> do we have the precedent of adding things into the exception's property bag when it makes sense to do so?

Not really.  You can see all of our own accesses to Exception.Data here:
https://source.dot.net/#System.Private.CoreLib/Exception.cs,33f0ea57f71b3a1e,references
and its very little from the core libs.
